### ISSUE TITLE
RxApp.SuspensionHost.ShouldPersistState fires twice upon sending the app to the background

### ISSUE THREAD

**Issue by RwSgxx** [reactiveui/ReactiveUI#1865]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

RxApp.SuspensionHost.ShouldPersistState fires twice upon sending the app to the background.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

Use SuspensionHost in an Android App and subscribe to RxApp.SuspensionHost.ShouldPersistState.

**What is the expected behavior?**

It should fire only once when sending the app to the background.

**What is the motivation / use case for changing the behavior?**

It is not expected to behave in this way. It can break the logic in the subscription and also perform unnecessary processing when the app is expected to go to the background.

**Which versions of ReactiveUI, and which platform / OS are affected by this issue? Did this work in previous versions of ReativeUI? Please also test with the [latest stable](https://www.nuget.org/profiles/reactiveui) and [development snapshot](https://reactiveui.net/docs/getting-started/installation/nuget-packages/development-package-feed)**

The App is a Xamarin Forms App running on Android. This doesn't happen on UWP.
ReactiveUI 9.4.1
Xamarin Forms 3.4.0.1008975

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**

**Comment by RLittlesII** [comment]
@RwSgxx Thanks for submitting this issue.  We will be looking into this.

**Comment by cabauman** [comment]
Old related issue: #991

As hinted tat in that issue, I feel that `_onSaveInstanceState` should not trigger _ShouldPersistState_ because _OnPause_ already handles that situation and more, quoted directly from [Android documentation](https://developer.android.com/reference/android/app/Activity):

> Note that it is important to save persistent data in onPause() instead of onSaveInstanceState(Bundle) because the latter is not part of the lifecycle callbacks, so will not be called in every situation as described in its documentation.

Here's the [code in question](https://github.com/reactiveui/ReactiveUI/blob/5cab37cb8e6726a731424d2e2447b047ae209016/src/ReactiveUI/Platforms/android/AutoSuspendHelper.cs#L54).

I'll make a PR unless someone has an objection.

**Comment by glennawatson** [comment]
@cabauman Go ahead and lodge a PR to fix both #1865 and #991.. Thanks.
